Sources/Commands/PackageTools/InstalledPackages.swift

```diff
@@ -0,0 +1,141 @@
+            case 1:
+                productToInstall = possibleCanidates[0].underlyingProduct
+            default: // More than one, check for possible
+                guard let product, let first = possibleCanidates.first(where: { $0.name == product }) else {
+                    throw StringError("Multiple canidates found, however, no product was specified. specify a product with the --product")
+                }
+                
+                productToInstall = first.underlyingProduct
+            }
+            
+            if let existingPkg = alreadyExisting.first(where: { $0.name == productToInstall.name }) {
+                throw StringError("\(productToInstall.name) is already installed at \(existingPkg.url)")
+            }
+            
+            try tool.createBuildSystem(explicitProduct: productToInstall.name).build(subset: .product(productToInstall.name))
+            
+            let binPath = try tool.buildParameters().buildPath.appending(component: productToInstall.name)
+            let finalBinPath = swiftpmBinDir.appending(component: binPath.basename)
+            try tool.fileSystem.copy(from: binPath, to: finalBinPath)
+            
+            let pkgInstance = InstalledPackageProduct(url: .init(finalBinPath))
+            alreadyExisting.append(pkgInstance)
+            
+            try JSONEncoder().encode(path: installedPackagesJSONPath, fileSystem: tool.fileSystem, alreadyExisting)
+        }
+    }
+    
+    struct Uninstall: SwiftCommand {
+        
+        static var configuration: CommandConfiguration {
+            CommandConfiguration(commandName: "experimental-uninstall", abstract: "Offers the ability to uninstall executable products of installed package products")
+        }
+        
+        @OptionGroup
+        var globalOptions: GlobalOptions
+        
+        @Argument(help: "Name of the executable to remove.")
+        var name: String
+        
+        func run(_ tool: SwiftTool) throws {
+            let dotSwiftPMDir = try tool.fileSystem.dotSwiftPM
+            var alreadyRegistered = (try? InstalledPackageProduct.registered(tool.fileSystem)) ?? []
+            
+            guard let whatWeWantToRemove = alreadyRegistered.first(where: { $0.name == name || $0.url.basename == name }) else {
+                // The installed executable doesn't exist - let the user know, and stop here.
+                var stringErrorToShowTheUser = "No such installed executable as \(name)"
+                
+                // and, in case there are any installed executables, let the user know which ones do exist.
+                if !alreadyRegistered.isEmpty {
+                    stringErrorToShowTheUser += ", existing installed executables: \(alreadyRegistered.map(\.name).joined(separator: "\n"))"
+                }
+                
+                throw StringError(stringErrorToShowTheUser)
+            }
+            
+            try tool.fileSystem.removeFileTree(whatWeWantToRemove.url)
```

Comment:
Let's make variable name more idiomatic and reflecting the fact that no "registration" is occurring:
```suggestion
            let alreadyInstalled = (try? InstalledPackageProduct.installedProducts(tool.fileSystem)) ?? []
            
            guard let removedExecutable = alreadyInstalled.first(where: { $0.name == name }) else {
                // The installed executable doesn't exist - let the user know, and stop here.
                var stringError = "No such installed executable as \(name)"
                
                // and, in case there are any installed executables, let the user know which ones do exist.
                if !alreadyInstalled.isEmpty {
                    stringError += ", existing installed executables: \(alreadyRegistered.map(\.name).joined(separator: "\n"))"
                }
                
                throw StringError(stringError)
            }
            
            try tool.fileSystem.removeFileTree(removedExecutable.path)
```